### PR TITLE
session, store: remove sessionBootstrapped

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve?

Issue Number: ref #57242

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: pkg/session/tidb.go [hunk 1]
```diff
@@ -148,21 +145,7 @@ var (
...
-	storeBootstrapped[storeUUID] = true
-}
-
-// unsetStoreBootstrapped delete store uuid from stored bootstrapped map.
-// currently this function only used for test.
-func unsetStoreBootstrapped(storeUUID string) {
-	storeBootstrappedLock.Lock()
-	defer storeBootstrappedLock.Unlock()
-	delete(storeBootstrapped, storeUUID)
+	store.SetOption(StoreBootstrappedKey, nil)
```
Comment (zimulala) [inline pkg/session/tidb.go:148]: Could we remove the comment on line145？

Comment (xhebox) [reply pkg/session/tidb.go:148]: I left it for next PR of removing domap.

File: pkg/session/session.go [hunk 2]
```diff
@@ -3892,10 +3892,8 @@ func mustGetStoreBootstrapVersion(store kv.Storage) int64 {
 }
 
 func getStoreBootstrapVersionWithCache(store kv.Storage) int64 {
-	storeBootstrappedLock.Lock()
```
Comment (xhebox) [inline pkg/session/session.go:3895]: Called in `cmd/tidb-server/main.go`, removing lock wont lead to race.

**Review by lance6716 — APPROVED** [review]

**Pull request merged**